CMakeLists.txt

```diff
@@ -412,16 +412,18 @@ set(MLPACK_LIBRARY_DIRS ${MLPACK_LIBRARY_DIRS} ${Boost_LIBRARY_DIRS})
 add_definitions(-DBOOST_TEST_DYN_LINK)
 
 # Detect OpenMP support in a compiler. If the compiler supports OpenMP, flags
-# to compile with OpenMP are returned and added and the HAS_OPENMP definition is
-# added for compilation.
+# to compile with OpenMP are returned and added and the HAS_OPENMP definition
+# is added for compilation.  Visual Studio doesn't support unsigned type index
+# in for loop, while OpenMP uses uword in for loop.  So to prevent build crash
+# on Windows, disable OpenMP, even though its support was detected.
```

Comment:
So to be more specific, the problem here actually arises only in Armadillo.  I don't see a need to disable OpenMP on Windows in general, but instead maybe we could have something like this?

```
// or _MSVC?  Other Windows compilers should be fine.
#ifdef _WIN32 && ARMA_USE_OPENMP

use arma::gmm_diag ...

#else 

Log::Warn << "Cannot use arma::gmm_diag on Visual Studio due to OpenMP compilation issues!"
    << " Using slower EMFit::Estimate() instead..." << std::endl;
// use EMFit::Estimate() to estimate a GMM using the current code with DiagonalConstraint

#endif 
```

What do you think?  I think that keeps the codebase relatively simple, and I don't mind if we have to use the slower implementation in just the Visual Studio case.  We can wait for Visual Studio to fix the issue (we will probably wait forever though).  I have some more comments about this below, so take a look at those and that might clarify what I mean a bit. :)

Reply:
Agreed, It looks good to me. As you mentioned, it keeps the codebase relatively simple. :)